pkg/blobstore/local/closable_block_list.go

```diff
@@ -0,0 +1,81 @@
+package local
+
+import (
+	"github.com/buildbarn/bb-storage/pkg/blobstore/buffer"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// errClosedForWriting is an error code to indicate that the BlockList does not
+// accept any more write operations and ongoing write operations will fail.
+var errClosedForWriting = status.Error(codes.Unavailable, "Closed for writing")
```

Comment:
Keep in mind that this error message gets sent back to the user. Maybe it's better to return something like: "Cannot write object to storage, as storage is shutting down"